File: java/interpreter/src/com/miwan/interpreter/lexical/Lexeme.java
package com.miwan.interpreter.lexical;

/**
 * @author liuziang
 * @contact liuziang@liuziangexit.com
 * @date 12/18/2019
 * <p>
 * 词法单元
 */

public class Lexeme {
	public Lexeme(String text, TokenKind kind) {
		this.text = text;
		this.kind = kind;
	}

	@Override
	public String toString() {
		return text;
	}

	public final String text;
	public final TokenKind kind;
}

File: cxx/interpreter/src/lexeme.cpp
#include <lexical/lexeme.h>

interpreter::lexeme::lexeme(const std::string &text,
                            interpreter::token_kind kind,
                            const interpreter::text_position &position)
    : text(text), kind(kind), position(position) {}

interpreter::text_position::text_position(const interpreter::text_position &) =
    default;

interpreter::text_position::text_position(uint32_t row, uint32_t col)
    : row(row), col(col) {}
